edge-modules/TestAnalyzer/Controllers/TestOperationResultController.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestAnalyzer.Controllers
+{
+    using System.IO;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Mvc;
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TestOperationResultController : Controller
+    {
+        [HttpPost]
+        public async Task<StatusCodeResult> PostAsync(TestOperationResult result)
+        {
+            LegacyTestOperationResult legacyResult = LegacyTestOperationResult.Convert(result);
+
+            switch (result.ResultType)
+            {
+                case TestResultType.LegacyDirectMethod:
+                    await ReportingCache.Instance.AddDirectMethodStatusAsync(legacyResult);
```

Comment:
Why can't these methods take TestOperationResult instead? It should not be too much work?

Reply:
Initially I can do it and see.